google-cloud-bigquery/lib/google/cloud/bigquery/query_job.rb

```diff
@@ -490,15 +511,18 @@ def wait_until_done!
         #
         def data token: nil, max: nil, start: nil
           return nil unless done?
-
+          if ddl? || dml?
```

Comment:
Checking if totalRows is nil or 0 will be more future-proof. The BigQuery team has at least 1 more DDL statement coming.

Reply:
https://github.com/googleapis/google-cloud-java/pull/3469

Reply:
The other thing about Java is that they wrap the underlying totalRows with an integer, so it's harder to disambiguate null versus 0, but I think we do want to avoid fetching rows in both cases.

Reply:
So which should we use here, 0 or nil?

Reply:
In my experience totalRows has been nil in the response for both DDL and DML queries, but I don't know if we should rely on that. I'd prefer checking for both, since in both cases we want to return empty results.

Reply:
Updated to `nil`.